### ISSUE TITLE
Cabal doesn't pass cc-options to GHC when compiling ordinary Haskell sources

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#9801]
... however with CApiFFI, GHC will invoke C-compiler which may end up compile a large non-trivial amount of code (e.g. `CApiFFI` is a way to use header only libraries, without creating a dummy source for them).

**Comment by phadej** [comment]
IMHO, `cc-options` (and ld-options and cpp-options and ...) should be always passed when invoking GHC, similarly as `ghc-options` should be always used when invoking ghc - regardless of what is the intention of a particular GHC-call. GHC might use or not use the options, Cabal cannot know.

**Comment by ulysses4ever** [comment]
@phadej could you also sketch the cabal file that should be used with this?

also, is this a dup of #4435?

**Comment by ulysses4ever** [comment]
> IMHO, cc-options (and ld-options and cpp-options and ...) should be always passed when invoking GHC

I agree. @zlonast, you recently did some work around passing of various options to various places. Would you be interested in taking a look at this issue? I imagine, having the context in your head might render this task trivial.

**Comment by zlonast** [comment]
> Would you be interested in taking a look at this issue? I imagine, having the context in your head might render this task trivial.

Yes, let me try to solve the issue 👀
